src/components/rule-builder/input-selection-box.component.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState, useRef, useCallback, useMemo } from 'react';
+import { TextInput, MultiSelect } from '@carbon/react';
+import { useDebounce } from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import styles from './input-selection-box.scss';
+
+interface CustomComboBoxProps {
+  id: string;
+  items: Array<Record<string, string>>;
+  onChange: (selectedItem: string | Array<{ concept: string; label: string }>) => void;
+  selectedAnswers: Array<{ concept: string; label: string }>;
+  value: string;
+  size: string;
+  isMultipleAnswers: boolean;
+  setSelectedAnswers: React.Dispatch<
+    React.SetStateAction<
+      Array<{
+        concept: string;
+        label: string;
+      }>
+    >
+  >;
+}
+
+const InputSelectionBox = React.memo(
+  ({ items, onChange, value, size, id, isMultipleAnswers, selectedAnswers }: CustomComboBoxProps) => {
+    const [inputValue, setInputValue] = useState<string>(value || '');
+    const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+    const [searchTermValue, setSearchTermValue] = useState<string>('');
+    const conditionalValue = useDebounce(inputValue, 500);
+    const selectedSearchValue = useDebounce(searchTermValue, 500);
+    const { t } = useTranslation();
+    const dropdownRef = useRef<HTMLUListElement>(null);
+
+    useEffect(() => {
+      if (searchTermValue.length) onChange(conditionalValue);
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
It seems dangerous not to include `searchTermValue.length` here? Sometimes ignoring deps in `useEffect()` makes sense, but that's usually where there's another `useEffect()` hook that applies the changes in the cases this mises.